benchmarks/nel/scripts/candidate_generation/base.py

```diff
@@ -34,27 +29,23 @@ def __call__(
         RETURNS (Iterator[Candidate]): Candidates for specified entity.
         """
 
-        if dataset_id not in self._pipelines:
-            self._pipelines[dataset_id] = spacy.load(
-                Dataset.assemble_paths(dataset_id)["nlp_base"]
-            )
+        if self._pipeline is None:
+            self._pipeline = spacy.load(Dataset.assemble_paths(dataset_id)["nlp_base"])
             with open(Dataset.assemble_paths(dataset_id)["entities"], "rb") as file:
                 self._entities[dataset_id] = pickle.load(file)
-        if dataset_id not in self._lookup_structs:
-            self._lookup_structs[dataset_id] = self._init_lookup_structure(
-                kb, max_n_candidates, **kwargs
-            )
+        if self._lookup_struct is None:
+            self._lookup_struct = self._init_lookup_structure(kb, max_n_candidates, **kwargs)
 
         # Retrieve candidates from KB via their aliases.
         return self._fetch_candidates(dataset_id, span, kb, max_n_candidates, **kwargs)
 
     @abc.abstractmethod
-    def _init_lookup_structure(
-        self, kb: KnowledgeBase, max_n_candidates: int, **kwargs
-    ) -> Any:
+    def _init_lookup_structure(self, kb: KnowledgeBase, max_n_candidates: int, **kwargs) -> Any:
         """Init container for lookups for new dataset. Doesn't do anything if initialized for this dataset already.
         kb (KnowledgeBase): KnowledgeBase containing all possible entity candidates.
         max_n_candidates (int): Max. number of candidates to generate.
+        similarity_threshold (float): Minimal similarity of candidates with mentions. All candidates with lower
+            similarity are ignored.
```

Comment:
Why has this docstring been added?

Reply:
Mistake, thanks for catching it. Removed in next commit.